apps/api/src/jobs/consumers/notifications/notification.consumer.ts

```diff
@@ -88,6 +88,12 @@ export abstract class NotificationConsumer {
           `Notification with ID ${notification.id} has attempted max allowed retries (sending), setting delivery status to ${DeliveryStatus.FAILED}`,
         );
         notification.deliveryStatus = DeliveryStatus.FAILED;
+        await this.notificationRepository.save(notification);
+        // Call webhook for all providers (skip and non skip) when delivery status is FAILED
+        await this.notificationQueueService.addNotificationToQueue(
```

Comment:
`await this.notificationRepository.save(notification);`

Were we not saving the details too earlier?

Reply:
- We did save details earlier as well. We save details in repository in `finally` block
- While testing without the save condition, saw mismatch in `delivery status`, It was marked as 4 (processing confirmation) instead of 6 (failed) that we just updated it to
- Before this change, we did not call webhook so updating repository at end worked
- Added save condition to update notification with latest `delivery status` for sending to webhook
- This syntax is followed everywhere we call webhook